src/main/java/org/jabref/gui/filelist/FileListDialogController.java

```diff
@@ -0,0 +1,157 @@
+        tfLink.textProperty().bindBidirectional(viewModel.linkProperty());
+    }
+
+    @FXML
+    void browseFileDialog(ActionEvent event) {
+
+        String fileText = viewModel.linkProperty().get();
+
+        Optional<Path> file = FileHelper.expandFilename(stateManager.getActiveDatabase().get(), fileText,
+                Globals.prefs.getFileDirectoryPreferences());
+
+        Path workingDir = file.orElse(Paths.get(Globals.prefs.get(JabRefPreferences.WORKING_DIRECTORY)));
+        String fileName = Paths.get(fileText).getFileName().toString();
+
+        FileDialogConfiguration fileDialogConfiguration = new FileDialogConfiguration.Builder()
+                .withInitialDirectory(workingDir)
+                .withInitialFileName(fileName).build();
+        Optional<Path> path;
+        if (showSaveDialog) {
+            path = dialogService.showFileSaveDialog(fileDialogConfiguration);
+        } else {
+            path = dialogService.showFileOpenDialog(fileDialogConfiguration);
+        }
+        path.ifPresent(newFile -> {
+            // Store the directory for next time:
+            Globals.prefs.put(JabRefPreferences.WORKING_DIRECTORY, newFile.toString());
+
+            // If the file is below the file directory, make the path relative:
+            List<Path> fileDirectories = this.stateManager.getActiveDatabase().get()
+                    .getFileDirectoriesAsPaths(Globals.prefs.getFileDirectoryPreferences());
+            newFile = FileUtil.shortenFileName(newFile, fileDirectories);
+
+            viewModel.linkProperty().set(newFile.toString());
+            tfLink.requestFocus();
+        });
+    }
+
+    @FXML
+    void cancel(ActionEvent event) {
+        getStage().close();
+    }
+
+    @FXML
+    void ok_clicked(ActionEvent event) {
+
+    }
+
+    private void setValues(LinkedFile entry) {
+        tfDescription.setText(entry.getDescription());
+        tfLink.setText(entry.getLink());
+
+        cmbFileType.getSelectionModel().clearSelection();
+        // See what is a reasonable selection for the type combobox:
+        Optional<ExternalFileType> fileType = ExternalFileTypes.getInstance().fromLinkedFile(entry, false);
+        if (fileType.isPresent() && !(fileType.get() instanceof UnknownExternalFileType)) {
+            cmbFileType.getSelectionModel().select(fileType.get());
+        } else if ((entry.getLink() != null) && (!entry.getLink().isEmpty())) {
+            checkExtension();
+        }
+    }
+
+    private void checkExtension() {
+        if (cmbFileType.getSelectionModel().isEmpty() && (!tfLink.getText().trim().isEmpty())) {
+
+            // Check if this looks like a remote link:
+            if (REMOTE_LINK_PATTERN.matcher(tfLink.getText()).matches()) {
+                Optional<ExternalFileType> type = ExternalFileTypes.getInstance().getExternalFileTypeByExt("html");
+                if (type.isPresent()) {
+                    cmbFileType.getSelectionModel().select(type.get());
```

Comment:
@tobiasdiez 
I would like to move this nethod and some others to the view Model, but I don't know how/or if I should expose the **selection model** of the combobox to the view model.

Reply:
You have a few options here:
1) Create a property in the view model and bind the selected item property to it, similar to the following code:
https://github.com/JabRef/jabref/blob/cbd6844183c15479ea2429d3962197e78b1494c5/src/main/java/org/jabref/gui/keyboard/KeyBindingsDialogController.java#L37-L41
But I'm not sure if using a bidirectional binding works here out of the box, probably you need to use https://github.com/JabRef/jabref/blob/bee715d03ab6fddcd48af5a5cf9e134ae458d521/src/main/java/org/jabref/gui/util/BindingsHelper.java#L75 or a similar helper method.

2) Just return the to-be-selected item from the method in the view model and then really select it here in the controller.

Option 1 is definitely the cleaner solution but maybe a bit harder to setup (but definitely worth the initial effort since a similar problem might appear in the future).